### PR TITLE
[INFINITY-1884] Make test.sh use pytest directly

### PR DESCRIPTION
**Pull request by benclarkwood**

### CODE REVIEW

File: test.sh [hunk 1]
```diff
@@ -1,80 +1,129 @@
...
+    pytest_m="$2"
+    shift # past argument
+    ;;
+    -k)
+    pytest_k="$2"
+    shift # past argument
+    ;;
+    -s)
+    security="strict"
+    shift # past argument
```
Comment (elezar) [inline test.sh]: Isn't this going to shift past the next argument?

File: test.sh [hunk 2]
```diff
@@ -1,80 +1,129 @@
...
+    $azure_args \
+    -e SECURITY=$security \
+    -e "PYTEST_K=$pytest_k" \
+    -e "PYTEST_M=$pytest_m" \
+    -e FRAMEWORK=$framework \
+    -e STUB_UNIVERSE_URL=$STUB_UNIVERSE_URL \
+    -v $(pwd):/build \
+    -v $ssh_path:/ssh/key \
+    -w /build \
+    michaelellenburg/dcos-commons:v0 \
```
Comment (elezar) [inline test.sh]: We could add the docker image as an argument too (with a sane default).

Comment (benclarkwood) [inline test.sh]: That seems like a reasonable thing to do.

File: test.sh (hunk 1 above)
Comment (benclarkwood) [inline test.sh]: Well, that would explain some things. (FWIW this is still a WIP, I extracted it out into a separate PR to get the base changes in)

cc @falseperfection

Comment (elezar) [inline test.sh]: (No problem with regards to the WIP. Was just looking through things, and noticed that. Thought it may cause some unexpected behaviour).

Comment (falseperfection) [inline test.sh]: Ha! Good catch, @elezar, copy/paste is so often my enemy.

**Comment by benclarkwood** [comment]
Dupe of #1296
